src/kernel/thread/thread_switch.c

```diff
@@ -33,13 +33,15 @@ static void thread_prepare_switch(struct thread *prev, struct thread *next) {
 	sched_start_switch(&next->schedee);
 }
 
+extern void thread_set_current(struct thread *t);
 void thread_context_switch(struct thread *prev, struct thread *next) {
 	thread_prepare_switch(prev, next);
 
 	/* Preserve initial semantics of prev/next. */
 	cpudata_var(saved_prev) = prev;
 	ADDR_SPACE_PREPARE_SWITCH();
 
+	thread_set_current(next);
```

Comment:
May be move under **context_switch**

Reply:
`context_switch()` is platform-dependent function, so in this case we will need to duplicate this function call. Also changing current thread in `thread_context_switch()` doesn't seem wrong.